tinygrad/lazy.py

```diff
@@ -207,11 +208,23 @@ def shuffle_and_prune_movement_ops(self, st: ShapeTracker, op: MovementOps, arg:
         return root.reshape(ret.st.shape)
     return ret
 
-  def reduce_op(self:LazyBuffer, op:ReduceOps, new_shape:Tuple[int, ...]) -> LazyBuffer:
+  def _reduce_op(self:LazyBuffer, op:ReduceOps, new_shape:Tuple[int, ...]) -> LazyBuffer:
     if self.shape == tuple(new_shape): return self
     srcs = _push_movement_ops((self,)) if SHUFFLE_MOVEMENT_OPS else (self,)
     return create_lazybuffer(self.device, ShapeTracker(new_shape), ReduceOps, LazyOp(op, srcs, new_shape), self.dtype)
 
+  def reduce_op(self:LazyBuffer, op:ReduceOps, new_shape:Tuple[int, ...]) -> LazyBuffer:
+    if prod(self.shape) // prod(new_shape) > 8192:
```

Comment:
Heuristic here because no need to split kernel for small reductions... but this line needs work.

1. It would be preferable to have this as the main codepath that gets tested when running tests like `test/test_ops.py', as obviously coverage would be better that way, but many of the tests their don't meet this threshold.
2. Preference is to only run this on GPU like devices, as it is about 15% slower on CPU from my testing, but TBD what the best check here is? `len(lang.gid)` not readily available in this context to my knowledge